packages/design-tokens/elm/Kaizen/Theme.elm

```diff
@@ -0,0 +1,29 @@
+module Kaizen.Theme exposing (Theme(..), defaultTheme, themeDecoder)
+
+import Json.Decode as Decode exposing (Decoder)
+
+
+type Theme
+    = Heart
+    | Zen
+
+
+defaultTheme =
+    Zen
+
+
+themeDecoder : Decoder Theme
+themeDecoder =
+    Decode.string
+        |> Decode.map
+            (\themeString ->
+                case themeString of
+                    "zen" ->
+                        Zen
+
+                    "heart" ->
+                        Heart
+
+                    _ ->
+                        defaultTheme
```

Comment:
Any reason for this fallback? I wonder if the decoder should `fail` when it receives an unexpected string value, and if we want to tolerate a missing/invalid value then we can use `Decode.maybe` for that:

```elm
decodeValue (maybe (field "theme" themeDecoder)) flags
```

Reply:
Thanks for the review Kevin!

I was a bit torn on this. I definitely didn't want it to fail, because I'm not very confident this would  be handled well by consumers. Falling back the last theme (in this case Zen) seems better than the page crashing or showing a generic error page. Also, I just wanted to make this as easy as possible for consumers (eg, not having to handle Maybes), as there will be a lot of friction in using the theme switcher in Elm so I'd be more worried people wouldn't find the time to implement it at all (which would guarantee no theme switching!). Even with these new features in Kaizen, there's still a lot of work to do to hook up the new illustrations - using the React `ThemeContext`, pass a flag to Elm and then prop drilling the theme all that way down to every Elm component. I happen to have a good amount of time this week so I can go deep with the prop drilling, etc, to get in our app but I'm not sure I can expect people in other teams with Elm codebases to do the same. 

Earlier as a much easier approach I suggested just immediately upgrading all the Elm illustrations to Heart, which would mean some customers not on the feature flag would see some components with Zen styles but Heart illustrations (because I thought implementing theme switching in Elm would be too much work for people). The Heart rollout team wasn't comfortable with this idea though.

There might be a much more magical solution for doing something like "context" in Elm to magically pass vars down the chain, but that would require a lot more investigation of feasibility.